02_bubble/pfannerj/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var mainout io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintf(mainout, "Bubble sort starting...\n")
+	fmt.Fprintln(mainout, bubble([]int{3, 2, 1, 5}))
+	fmt.Fprintf(mainout, "Bubble sort completed...\n")
+	fmt.Fprintf(mainout, "Insertion sort starting...\n")
+	fmt.Fprintln(mainout, insertion([]int{3, 2, 5, 1}))
+	fmt.Fprintf(mainout, "Insertion sort completed...\n")
+}
+
+func bubble(s []int) []int {
+	sCopy := make([]int, len(s))
+	copy(sCopy, s)
+	var saveNbr int
+	swapped := false
+	i := 0
+	for {
+		if (i + 1) >= len(s) {
+			i = 0
+			if swapped {
+				swapped = false
+			} else {
+				break //Sort is complete if no values were 'swapped' during the previous iteration
+			}
+		}
+		if sCopy[i] > sCopy[i+1] {
+			saveNbr = sCopy[i]
+			sCopy[i], sCopy[i+1] = sCopy[i+1], saveNbr
+			swapped = true
+		}
+		i++
+	}
+	return sCopy
+}
+
+func insertion(s []int) []int {
+	sCopy := make([]int, len(s))
+	copy(sCopy, s)
+	var moveNbr int
+	var saveNbr int
```

Comment:
Multiple variables of the same type can be declared in one line. `var movNbr, saveNbr int`. Also please give meaningful variable names.

Reply:
yep, will change to a single line... but I need some suggestions on what you think would be more meaningful variable names... they are meaningful to me...

Reply:
I am not sure what 'Nbr' means.

Reply:
Oh, ok... maybe this is my old-fashioned coding brain coming out... but when I was at Uni it was drilled into us to use Nbr as an abbreviation for 'Number'... as opposed to using 'No' which can be confused with No/Yes.

Reply:
Yes, Haha. I honestly had no idea what Nbr meant.

Reply:
what would u use?

Reply:
I think either saveNum or 'saveNumber' is fine. Extra characters won't take more memory.